### PR TITLE
Use httr2 cache

### LINKED ISSUE #203: Use a httr2 cache?
https://docs.github.com/en/rest/using-the-rest-api/best-practices-for-using-the-rest-api?apiVersion=2022-11-28#use-conditional-requests-if-appropriate

### PR DESCRIPTION
**Pull request by hadley**
This is really simple to implement but I think it probably needs some way to suppress and it might cause problems with `R CMD check`. Maybe an option?

Fixes #203

### CODE REVIEW

**Comment by gaborcsardi** [comment]
I think we can have an option to opt out, and we don't run any examples or tests on CRAN, so that part is fine. OTOH tests are failing now.

**Comment by hadley** [comment]
The failing tests are why you shouldn't develop HTTP based packages on a plane 😄. I'll fix them tomorrow.

File: R/gh.R [hunk 1]
```diff
@@ -287,7 +287,13 @@ gh_make_request <- function(x, error_call = caller_env()) {
   }
   req <- httr2::req_headers(req, !!!x$headers)
 
-  req <- httr2::req_cache(req, path = tools::R_user_dir("gh", "cache"))
+  if (!isFALSE(getOption("gh_cache"))) {
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/gh.R:290]: I thought we used `_` now?

[the remark this answers is not in the document]
Comment (gaborcsardi) [reply R/gh.R:290]: OK then, I missed that.

**Comment by gaborcsardi** [comment]
Thanks!

**Pull request merged**